plugins/panels/similarity_search/operators.py

```
@@ -0,0 +1,260 @@
+            run_data["result_ids"] = result_ids
+            run_data["result_count"] = len(result_ids)
+            run_data["result_view"] = result_view_stages
+            run_data["end_time"] = datetime.now(timezone.utc).isoformat()
+            manager.set_run(run_id, run_data)
+
+            ctx.set_progress(1.0, label="Done")
+
+            return {"run_id": run_id}
+
+        except Exception as e:
+            logger.error(
```

Ruff (0.15.5)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)